api/requests/permissions.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from rest_framework import permissions as drf_permissions
+
+from api.base.utils import get_user_auth
+from osf.models.action import NodeRequestAction
+from osf.models.mixins import RequestableMixin
+from osf.models.node import Node
+from osf.utils.workflows import DefaultTriggers
+from website.util import permissions as osf_permissions
+
+
+class NodeRequestPermission(drf_permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        auth = get_user_auth(request)
+        if auth.user is None:
+            return False
+
+        target = None
+        trigger = None
```

Comment:
I don't think you need to define this here - if trigger isn't set in any of the below if/elif statements it'll raise a `ValueError` anyhow